### PR TITLE
Track active migrations in Prometheus and `tctl top`

### LINKED ISSUE #12365: improve data migration visibility
### What would you like Teleport to do?

Teleport does migrations sometimes. We should communicate the status of these to a teleport administrator so they can effectively plan and execute upgrades.

They should be able to refer to our documentation to see whether a migration for a given from and to version would possibly be required.

There should be information about migration status in a metrics or healthcheck endpoint so systems can be built to automate upgrades.

### What problem does this solve?

* Currently it is difficult-to-impossible to tell whether a given version upgrade will have any migrations.
* At startup, it is not clear how to tell whether migrations were determined to be needed, whether they are currently running, or what the result of the migration might be.

### If a workaround exists, please include it.

Currently, one would need to do some significant looking through the codebase to determine how migrations work to try to anticipate whether they will be needed or not.

gz#4254

### PR DESCRIPTION
**Pull request by vitorenesduarte**
Fixes #12365.

This commit adds a new Prometheus gauge `teleport_migrations` that tracks for each migration if it is active (1) or not (0).

This gauge is then leveraged in `tctl top` to show a set of active migrations.

#### Testing

- `http://127.0.0.1:3000/metrics`:

```
# HELP teleport_migrations Migrations tracks for each migration if it is active (1) or not (0).
# TYPE teleport_migrations gauge
teleport_migrations{migration="db_authority"} 1
teleport_migrations{migration="remote_clusters"} 1
```

- `tctl top`:

Screenshot 2022-12-20 at 15 26 43

### CODE REVIEW

File: tool/tctl/common/top_command.go [hunk 1]
```diff
@@ -879,6 +883,24 @@ func getRemoteClusters(metric *dto.MetricFamily) []RemoteCluster {
...
+		return nil
+	}
+	var out []string
+	for _, counter := range metric.Metric {
+		if counter.Gauge.GetValue() == 0 {
+			continue
+		}
+		for _, label := range counter.Label {
+			if label.GetName() == "name" {
+				out = append(out, label.GetValue())
```
Comment (probakowski) [inline tool/tctl/common/top_command.go:897]: can there be more than one `name` label? If not we can break here

**Review by probakowski — APPROVED** [review]
Looks good, one small question

File: tool/tctl/common/top_command.go (hunk 1 above)
Comment (vitorenesduarte) [reply tool/tctl/common/top_command.go:897]: Thanks! Done in f26f8731a0edc2b1ad7eb0e72d651c82a3be57bc.

**Review by tobiaszheller — APPROVED** [review]

File: lib/auth/init.go [hunk 2]
```diff
@@ -551,6 +551,20 @@ func shouldInitReplaceResourceWithOrigin(stored, candidate types.ResourceWithOri
 	return false, nil
 }
 
+// migrationStart marks the migration as active.
+// It should be called when a migration starts.
+func migrationStart(ctx context.Context, migrationName string) {
+	log.Debugf("Migrations: %q migration started.", migrationName)
+	migrations.WithLabelValues(migrationName).Set(float64(1))
```
Comment (tigrato) [inline lib/auth/init.go:558]: I don't think you need this typecast

**Review by tigrato — APPROVED** [review]

File: lib/auth/auth.go [hunk 3]
```diff
@@ -384,10 +384,18 @@ var (
 		[]string{teleport.TagVersion},
 	)
 
+	migrations = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
```
Comment (espadolini) [inline lib/auth/auth.go:388]: Add a `Namespace: teleport.MetricNamespace` to this (and `teleport_` to metrics.mdx).

**Review by espadolini — CHANGES_REQUESTED** [review]
LGTM other than the lack of `teleport_` namespace for a new metric.

Feel free to dismiss this CR once that's addressed.

File: lib/auth/init.go (hunk 2 above)
Comment (vitorenesduarte) [reply lib/auth/init.go]: Thanks! Removed in 3083c9b8170cc8e8e319d2d6d896c59da45789bd.

File: lib/auth/auth.go (hunk 3 above)
Comment (vitorenesduarte) [reply lib/auth/auth.go:388]: Done in 21b2303ccdf26d0204203d41964d2430171a0055.

**Review by espadolini — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**